site-service/src/main/java/uk/ac/ox/ndph/mts/site_service/controller/SiteController.java

```diff
@@ -37,22 +38,20 @@ public SiteController(SiteService siteService) {
         return ResponseEntity.status(HttpStatus.CREATED).body(new Response(siteId));
     }
 
-    @PreAuthorize("@authorisationService.authorise('view-site')")
-    @PostAuthorize("@authorisationService.filterUserSites(returnObject.getBody(), #role)")
+    @PostAuthorize("@siteServiceImpl.filterUserSites(returnObject, #role, 'view-site')")
     @GetMapping
-    public ResponseEntity<List<Site>> getSites(@RequestParam(value = "role", required = false) String role) {
-        return ResponseEntity.status(HttpStatus.OK).body(siteService.findSites());
-    }
-
-    @GetMapping("/assigned")
-    @PostAuthorize("@authorisationService.filterUserSites(returnObject.getBody(), null)")
-    public ResponseEntity<List<Site>> unauthorizedSites(@RequestParam(value = "role", required = false) String role) {
-        return ResponseEntity.status(HttpStatus.OK).body(siteService.findSites());
+    public List<SiteDTO> getSites(@RequestParam(value = "role", required = false) String role) {
+        return siteService.findSites();
     }
 
     @GetMapping("/{id}")
     public ResponseEntity<Site> findSiteById(@PathVariable String id) {
         return ResponseEntity.ok(siteService.findSiteById(id));
     }
 
+    @GetMapping("/parents/{id}")
```

Comment:
If I understand this correctly, it's deliberate that this method does not have any authz on it and important, since it is called by the authz component and we otherwise get into circularity?  If so should there be a comment here to tell future developers that this method is special?